autoscale/src/main/java/com/sequenceiq/periscope/repository/ScalingActivityRepository.java

```diff
@@ -45,6 +45,9 @@ List<ScalingActivity> findAllByClusterAndActivityStatusBetweenInterval(@Param("c
     List<ScalingActivity> findAllByClusterBetweenInterval(@Param("clusterId") Long clusterId, @Param("startTimeFrom") Date startTimeFrom,
             @Param("startTimeUntil") Date startTimeUntil);
 
+    @Query("SELECT st.id FROM ScalingActivity st WHERE st.endTime <= :endTimeBefore")
```

Comment:
This query will be executed every 10 minutes, and will not be a cheap query. endtime vs starttime? If possible, a status filter?

Reply:
Updated the query used here

Reply:
Looks like the query being used changed in the new patch, and there's some questions on that in the review. Still need to look at the efficiency of the query once it is finalized.